### ISSUE TITLE
Add id and aria-label attributes to photoswipe modal on single product page

### ISSUE THREAD

**Issue by Manussakis** [woothemes/woocommerce#59953]
### Prerequisites (mark completed items with an [x]):

- [x] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [x] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

### Issue Description:
The buttons responsible for opening the photoswipe modal on a product page must have the `aria-controls` attribute whose value is the same as the photoswipe `id`. However, the modal doesn't have the `id` attribute.

The modal also needs to have an accessible name so assistive technologies can properly announce its purpose.

### Screenshots

<img width="1908" height="887" alt="Image" src="https://github.com/user-attachments/assets/ecea75bf-f468-46d6-8004-7839708ca956" />

### Component or Page:
Pages: Single Product

### Relevant code snippets:
plugins/woocommerce/templates/single-product/photoswipe.php

### Recommended fix:
- **`id` attribute**:
Add a unique `id` attribute to the photoswipe modal. Other elements, responsible for triggering the modal, will use the `id` value as their `aria-controls` attribute.
- **`aria-label` attribute**:
Add an accessible name to the dialog: a short, descriptive label that screen readers and other assistive technologies use to identify the dialog's purpose.

**Comment by PanosSynetos** [comment]
Hey @Manussakis - There's already an issue open by @amberhinds  and @sunyatasattva is assigned to it.
 
- https://github.com/woocommerce/woocommerce/issues/59746

Since you've self-assigned this, I'll close the other issue. Please tag @PanosSynetos @sunyatasattva @amberhinds when you create the PR so we're aware and tag me as a reviewer, as I need this fix in some of the extensions we maintain.

Thank you!

**Comment by PanosSynetos** [comment]
@Manussakis please leave @sunyatasattva assigned to this issue too, so we can track this internally
